src/result/error.rs

```diff
@@ -40,11 +40,11 @@ impl Error {
         Self { code, info }
     }
 
-    // Use internally to create an Error object without error info. Typically used with QueryInterface
-    // (E_NOINTERFACE) or the absense of an object to return (E_POINTER) to avoid the code gen overhead
-    // for casts that should be cheap (few instructions). Think of it as a way to create recoverable errors
-    // that don't need th overhead of debugging origination info.
-    pub fn fast_error(code: HRESULT) -> Self {
+    /// Use internally to create an Error object without error info. Typically used with [`Interface::query`]
+    /// ([`HRESULT::E_NOINTERFACE`]) or the absence of an object to return ([`HRESULT::E_POINTER`]) to avoid the
+    /// code gen overhead for casts that should be cheap (few instructions). Think of it as a way to create
+    /// recoverable errors that don't need the overhead of debugging origination info.
+    pub const fn fast_error(code: HRESULT) -> Self {
```

Comment:
Not sure if this should be made const just yet, but it allows to have nice things like:

```rust
impl Error {
    pub const E_POINTER: Self = Self::fast_error(HRESULT::E_POINTER);
    pub const E_NOINTERFACE: Self = Self::fast_error(HRESULT::E_NOINTERFACE);
}
```

Reply:
No, I'm aggressively using the latest versions of Rust to take advantage of new features (like `const`). Once the Windows crate reaches 1.0 I'll probably start supporting a min version.

Reply:
Nice!

In other words, this could have been a welcome change if it weren't for not really wanting to reexport these constants. I guess we can still annotate all applicable functions as `const` for versatility.

Reply:
Yep, feel free to suggest other improvements based on the bleeding edge of Rust. 😉